### PR TITLE
UniversalRP ShaderPreprocessor can strip gbuffer related variants if no DeferredRenderer is used

### PR DESCRIPTION
**Pull request by kaychang-unity**
# Purpose of this PR
Remove gbuffer variants if no DeferredRenderer is listed in any Scriptable Pipeline assets.
This saves a bit of build time (number of gbuffer variants is actually small).

# Testing status
## Manual Tests
Tested all gbuffer variants can be stripped out in UniversalGraphicsTests.

## Links
**Yamato**: (Select your branch) https://yamato.prd.cds.internal.unity3d.com/jobs/902-Graphics
> Any test projects or documents to go with this to help reviewers?

# Comments to reviewers
> Notes for the reviewers you have assigned.

### CODE REVIEW

**Comment by phi-lira** [comment]
Rerunning OSX_Metal_playmode.

File: com.unity.render-pipelines.universal/Runtime/DeferredRenderer.cs [hunk 1]
```diff
@@ -71,6 +71,8 @@ public sealed class DeferredRenderer : ScriptableRenderer
         Material m_TileDeferredMaterial;
         Material m_StencilDeferredMaterial;
 
+        public bool AccurateGbufferNormals { get { return m_DeferredLights.AccurateGbufferNormals; } }
```
Comment (phi-lira) [inline com.unity.render-pipelines.universal/Runtime/DeferredRenderer.cs:74]: add public API docs for any new public API, or make it internal.

Comment (kaychang-unity) [reply com.unity.render-pipelines.universal/Runtime/DeferredRenderer.cs]: @phi-lira I made the function internal.

**Comment by kaychang-unity** [comment]
@phi-lira Ready to land!

**Comment by phi-lira** [comment]
LGTM. Will let ABV finish just for safety.

**Comment by phi-lira** [comment]
These failures were caused by a regression in master. The regression is already fixed. Can you re-merge master into your branch and rerun jobs?

**Comment by kaychang-unity** [comment]
@phi-lira I merged again but ABV in Master was still red.
Looking at latest ABV from Master branch today, it is still red.

**Comment by kaychang-unity** [comment]
@phi-lira @ellioman I merged again with Master, but Master ABV is still broken AFAIK.

**Comment by kaychang-unity** [comment]
@phi-lira @ellioman  ABV is green :)

**Pull request merged**